Assets/Scripts/Settings/TimeSettings.cs

```diff
@@ -0,0 +1,34 @@
+﻿
+using Assets.Scripts.Gamemode;
+using System;
+
+namespace Assets.Scripts.Settings
+{
+    public class TimeSettings
+    {
+       
+        public float? currentTime { get; set; }
+        public float? dayLength { get; set; }
+        public bool? pause { get; set; }
+        public DateTime LastModified { get; set; }
+
+        public TimeSettings() { }
+        public TimeSettings(Difficulty difficulty)
+        {
+            switch (difficulty)
+            {
+                case Difficulty.Easy:
+                case Difficulty.Normal:
+                case Difficulty.Hard:
+                case Difficulty.Abnormal:
+                case Difficulty.Realism:
+                    dayLength = 1000;
```

Comment:
All settings should be assigned a value, otherwise you will get NullReferenceExceptions